astropy/coordinates/baseframe.py

```diff
@@ -867,28 +864,17 @@ def representation_info(self):
         return self._get_representation_info()
 
     def get_representation_component_names(self, which="base"):
-        out = {}
-        repr_or_diff_cls = self.get_representation_cls(which)
-        if repr_or_diff_cls is None:
-            return out
-        data_names = repr_or_diff_cls.attr_classes.keys()
-        repr_names = self.representation_info[repr_or_diff_cls]["names"]
-        for repr_name, data_name in zip(repr_names, data_names):
-            out[repr_name] = data_name
-        return out
+        cls = self.get_representation_cls(which)
+        if cls is None:
+            return {}
+        return dict(zip(self.representation_info[cls]["names"], cls.attr_classes))
 
     def get_representation_component_units(self, which="base"):
-        out = {}
         repr_or_diff_cls = self.get_representation_cls(which)
         if repr_or_diff_cls is None:
-            return out
+            return {}
         repr_attrs = self.representation_info[repr_or_diff_cls]
-        repr_names = repr_attrs["names"]
-        repr_units = repr_attrs["units"]
-        for repr_name, repr_unit in zip(repr_names, repr_units):
-            if repr_unit:
-                out[repr_name] = repr_unit
-        return out
+        return {k: v for k, v in zip(repr_attrs["names"], repr_attrs["units"]) if v}
```

Comment:
```suggestion
        return {k: v for k, v in zip(repr_attrs["names"], repr_attrs["units"], strict=True) if v}
```

Reply:
Py 3.10+ (https://peps.python.org/pep-0618/)

Reply:
There are situations where the `strict` argument is very useful, but explicitly specifying it everywhere sounds too verbose to me. I can't tell if using it here is helpful or not.

Reply:
I think it is very unlikely this would ever fail, as these are set at class definition time.

Reply:
> these are set at class definition time.

I guess your point can be interpreted as an argument both ways, but to me it sounds more like `strict=True` is correct (hence desirable) here.

Reply:
I think class definition might ensure this is right, but didn't check - to me, it seems overkill.

Reply:
I'm inclined to agree with @mhvk that the verbosity of having to specify `strict=True` here is not worth it.